### PR TITLE
Fix: Missing or Default `null` `@cypher` Arguments 

### PR DESCRIPTION
**Pull request by dmoree**
# Description

Proposed fix for #444 and #595 wherein nullable arguments of the cypher directive that are either missing or have a default value of `null` are not passed as parameters into the `apoc` procedure thereby causing a runtime error to occur.

This PR adds an arbitrary key-value pair of `{ [argument]: null }` to the arguments of the resolve tree in order to generate the correct `apocParam.strs` used in the `apoc` procedure generated by either a top-level query or projection.

# Issue

- #444 
- #595 

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [x] Integration tests have been updated
- [x] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

**Comment by darrellwarde** [comment]
Please ignore the ESLint violations being reported by Reviewdog - we're having a play with the ESLint rules for this as we speak.

File: packages/graphql/src/schema/resolvers/cypher.ts [hunk 1]
```diff
@@ -153,13 +153,25 @@ export default function cypherResolver({
...
+        const nullArgumentValues = field.arguments.reduce(
+            (res, argument) => ({
+                ...res,
+                ...{ [argument.name.value]: null },
+            }),
+            {}
+        );
+
+        // Overwrite null arguments with those defined in resolve tree
+        const argumentsWithNullValues = { ...nullArgumentValues, ...resolveTree.args };
```
Comment (darrellwarde) [inline packages/graphql/src/schema/resolvers/cypher.ts:168]: I don't know why, but I find this variable name very confusing, it sounds like "all of the arguments in this object have a null value". Maybe just `arguments` if that's not already declared in this scope?

File: packages/graphql/src/translate/create-projection-and-params.ts [hunk 2]
```diff
@@ -265,7 +265,21 @@ function createProjectionAndParams({
...
+            const nullArgumentValues = cypherField.arguments.reduce(
+                (r, argument) => ({
+                    ...r,
+                    ...{ [argument.name.value]: null },
+                }),
+                {}
+            );
+
+            // Overwrite null arguments with those defined in resolve tree
+            const argumentsWithNullValues = { ...nullArgumentValues, ...field.args };
```
Comment (darrellwarde) [inline packages/graphql/src/translate/create-projection-and-params.ts:280]: Same comment on this variable naming as in the other file.

**Review by darrellwarde — CHANGES_REQUESTED** [review]
Brilliant, this will make a lot of people very happy. Just a small comment about the naming of a single variable, but generally looks great!

It would be even better if we could see a little integration test of this working so that we can have 100% confidence that this works end-to-end if you don't mind?

File: packages/graphql/src/schema/resolvers/cypher.ts (hunk 1 above)
Comment (dmoree) [reply packages/graphql/src/schema/resolvers/cypher.ts:168]: Sorry, it does look confusing. Unfortunately, `arguments` is a reserved word in javascript and `args` is taken. Perhaps `apocArgs`? I do agree that it should be changed to be clearer.

It is only ever used in generating the `apocParams`. We could skip the entire problem and put it directly in `Object.entries` and let the comment above `nullArgumentValues` do the work.

Comment (darrellwarde) [reply packages/graphql/src/schema/resolvers/cypher.ts:168]: Of course it is, easy to forget all of the reserved words!

I think just pop it directly in `Object.entries` given that it's only spreading a couple of objects and not a huge multiline function. 🙂

Comment (dmoree) [reply packages/graphql/src/schema/resolvers/cypher.ts]: Fixed in https://github.com/neo4j/graphql/pull/769/commits/b27b37efc39a595019b8668b235737d14b7d6b65

**Comment by dmoree** [comment]
Do you mean TCK tests? Integration tests were added in https://github.com/neo4j/graphql/pull/769/commits/2a4c0be4452873f20f65747ff06a88df606369e7

**Comment by darrellwarde** [comment]
> Do you mean TCK tests? Integration tests were added in [2a4c0be](https://github.com/neo4j/graphql/commit/2a4c0be4452873f20f65747ff06a88df606369e7)

It's quite possible that I'm either blind or was just very confused when reviewing multiple PRs yesterday - let me take another look following the removal of that confusing variable. 🙂

File: packages/graphql/src/translate/create-projection-and-params.ts [hunk 3]
```diff
@@ -265,7 +265,21 @@ function createProjectionAndParams({
...
+
+            const apocParams = Object.entries(argumentsWithNullValues).reduce(
```
Comment (darrellwarde) [inline packages/graphql/src/translate/create-projection-and-params.ts:282]: Looks to me that this one still needs moving down into `Object.entries`?

```suggestion
            const apocParams = Object.entries({ ...nullArgumentValues, ...field.args }).reduce(
```

**Review by darrellwarde — CHANGES_REQUESTED** [review]
Just the other `argumentsWithNullValues` needs removing and then I think this is good to go - I added it as a suggestion so should just be able to commit from the UI. 🙂  I don't know why I missed the integration test - I think I'm losing it!

**Review by darrellwarde — APPROVED** [review]
Great, this looks good to me! 😄

**Pull request merged**